README.md

```diff
@@ -2,11 +2,876 @@
+
+## Manual Instrumentation with Java SDK
+
+### Add OpenTelemetry to FirstService and SecondService
+
+Required dependencies and configurations for FirstService and SecondService projects can be found [here](#setup-for-manual-instrumentation).
+
+### FirstService
+
+1. Ensure OpenTelemetry dependencies are included
+2. Ensure an OpenTelemetry Tracer is configured
+
+3. Ensure a Spring Boot main class was created by the Spring initializer
+
+```java
+@SpringBootApplication
+public class FirstServiceApplication {
+
+  public static void main(String[] args) throws IOException {
+    SpringApplication.run(FirstServiceApplication.class, args);
+  }
+}
+```
+
+4. Create a RestController for FirstService
+5. Create a span to wrap the FirstServiceController
+
+```java
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+
+import HttpUtils;
+
+@RestController
+@RequestMapping(value = "/message")
+public class FirstServiceController {
+  @Autowired
+  private Tracer tracer;
+
+  @Autowired
+  HttpUtils httpUtils;
+
+  private static String SS_URL = "http://localhost:8081/time";
+
+  @GetMapping
+  public String firstTracedMethod() {
+    Span span = tracer.spanBuilder("message").startSpan();
+    span.addEvent("Controller Entered");
+    span.setAttribute("what.are.you", "Je suis attribute");
+
+    try (Scope scope = tracer.withSpan(span)) {
+      return "Second Service says: " + httpUtils.callEndpoint(SS_URL);
+    } catch (Exception e) {
+      span.setAttribute("error", e.toString());
+      span.setAttribute("error", true);
+      return "ERROR: I can't tell the time";
+    } finally {
+      span.end();
+    }
+  }
+}
+```
+
+6. Configure `HttpUtils.callEndpoint` to inject span context into request. This is key to propagate the trace to the SecondService
+
+HttpUtils is a helper class that injects the current span context into outgoing requests. This involves adding the tracer id and the trace-state to a request header. For this example, I used `RestTemplate` to send requests from `FirstService` to `SecondService`. A similar approach can be used with popular Java Web Clients such as [okhttp](https://square.github.io/okhttp/) and [apache http client](https://www.tutorialspoint.com/apache_httpclient/apache_httpclient_quick_guide.htm). The key to this implementation is to override the put method in `HttpTextFormat.Setter<?>` to handle your request format. `HttpTextFormat.inject` will use this setter to set `traceparent` and `tracestate` headers in your requests. These values will be used to propagate your span context to external services.
```

Comment:
[Note] Make it clear who provides it: Spring/OT/Boot/You

Reply:
Also, if this is something reusable, it may be worth putting it in the repository with a link so people can clone it, or as a jar.

Reply:
Once I get the readme approved my plan is to pull all the sample code in an example project and have links throughout this tutorial.